src/core/dense_set.cc

```diff
@@ -197,6 +197,47 @@ bool DenseSet::Equal(DensePtr dptr, const void* ptr, uint32_t cookie) const {
   return ObjEqual(dptr.GetObject(), ptr, cookie);
 }
 
+void DenseSet::CloneBatch(unsigned len, CloneItem* items, DenseSet* other) const {
+  while (len) {
+    unsigned dest_id = 0;
+    for (unsigned i = 0; i < len; ++i) {
```

Comment:
I'm not sure regarding this, but it looks like items[i] can contain obj == nullptr and link == nullptr so maybe we can start not from 0 but from position that definitely not nullptr

Reply:
it can not be, because if link is null we do not increase dest_id, so the next iteration we will go over less items.
we always iterate over dense range